packages/@aws-cdk/aws-amplify-alpha/lib/app.ts

```diff
@@ -224,6 +225,8 @@ export class App extends Resource implements IApp, iam.IGrantable {
 
   constructor(scope: Construct, id: string, props: AppProps) {
     super(scope, id);
+    // Enhanced CDK Analytics Telemetry
+    addConstructMetadata(this, props);
```

Comment:
The majority of the code changes are auto generated so you'll see hundreds of `addConstructMetadata` method call across different L2 resources.

This method comes from this change https://github.com/aws/aws-cdk/pull/33232/files#diff-81f821b1205e7040fc3103bf7c0114060a6d5c43ebd2994aa4ed5906e42c9c5fR33. The main code change that needs to be reviewed is in `packages/aws-cdk-lib/core` as well as `tools/@aws-cdk/construct-metadata-updater`

Reply:
We only support collecting metadata from the constructor for now .. right ?

Reply:
This PR only supports constructor collection. I'll work on the method collections but I don't think that blocks us reviewing and merging this PR.